### PR TITLE
My Commit for the Issue #2864 Pascal's Triangle

### PR DESCRIPTION
**Pull request by iamhardikat11**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by iamhardikat11** [comment]
Made my PR. Please Review and Respond.

File: src/main/java/com/thealgorithms/maths/PascalTriangle.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                if (line == i || i == 0)
+                    arr[line][i] = 1;
+                    // The rest elements are sum of values just above and left of above
+                else
+                    arr[line][i] = arr[line-1][i-1] + arr[line-1][i];
+                System.out.print(arr[line][i]+" ");
+            }
+            System.out.println();
+        }
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/maths/PascalTriangle.java:72]:
```suggestion
    public static int[][] pascal(int n)
    {
        /**
         * @param arr  An auxiliary array to store generated pascal triangle values
         * @return
         */
        int[][] arr = new int[n][n];


        /**
         * @param line Iterate through every line and print integer(s) in it
         * @param i Represents the column number of the element we are currently on
         */
        for (int line = 0; line < n; line++)
        {
            /**
             *  @Every line has number of integers equal to line number
             */
            for (int i = 0; i <= line; i++)
            {
                // First and last values in every row are 1
                if (line == i || i == 0)
                    arr[line][i] = 1;
                    // The rest elements are sum of values just above and left of above
                else
                    arr[line][i] = arr[line-1][i-1] + arr[line-1][i];
            }
        }
        
        return arr;
    }
```

File: src/main/java/com/thealgorithms/maths/PascalTriangle.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+     *    1   7  21  35   35  21  7   1
+     *  1   8  28  56  70   56   28  8   1
+     *
+     */
+
+    public static void main (String[] args) {
+        Scanner sc = new Scanner(System.in);
+        int n = sc.nextInt();
+        printPascal(n);
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/maths/PascalTriangle.java:40]: Please remove the main() method and add some automated tests as was done in https://github.com/TheAlgorithms/Java/pull/2869/files, thanks!

**Review by siriak — CHANGES_REQUESTED** [review]
I've left a small comment about automated tests, everything else looks good!

**Review by siriak — DISMISSED** [review]
The build is failing because of a typo in the last test at com.thealgorithms.maths.ArrayLeftRotationTest.testForEight(PascalTriangleTest.java:38). Please fix it, everything else is solid, thanks!

**Comment by iamhardikat11** [comment]
This task (Build/build(pull_request)) is failing everytime I try, what can be done?

**Comment by iamhardikat11** [comment]
I have resolved all Checks please Merge my PR.

**Review by siriak — APPROVED** [review]
LGTM, thanks!

**Pull request merged**